### PR TITLE
FIX: Update to Glides Ingestion before Deploy

### PR DESCRIPTION
**Pull request by mzappitello**
There are a few issues with glides ingestion that need to be ironed out before deploying to production.

* Operator Sign Ins had a mislabeled field that wasn't being picked up.
* Add try / except blocks and clear logging around kinesis reader and the ingest records blocks.

### CODE REVIEW

File: src/lamp_py/aws/kinesis.py [hunk 1]
```diff
@@ -95,36 +95,42 @@ def get_records(self) -> List[Dict]:
...
-                    all_records.append(json.loads(record["Data"]))
-
-                if response["MillisBehindLatest"] == 0:
-                    break
-
-            # thrown if the shard iterator has expired. catch it and regenerate
-            # a new shard iterator.
-            except self.kinesis_client.exceptions.ExpiredIteratorException:
+
+        try:
```
Comment (mzappitello) [inline src/lamp_py/aws/kinesis.py:100]: reading records is under a try except with logging.

File: src/lamp_py/ingestion/glides.py [hunk 2]
```diff
@@ -97,29 +98,31 @@ def append_records(self) -> None:
...
-                end = start + relativedelta(months=1)
-                if end < now:
-                    table = joined_ds.filter(
-                        (pc.field("time") >= start) & (pc.field("time") < end)
-                    ).to_table()
-                else:
-                    table = joined_ds.filter(
-                        (pc.field("time") >= start)
-                    ).to_table()
+        with tempfile.TemporaryDirectory() as tmp_dir:
```
Comment (mzappitello) [inline src/lamp_py/ingestion/glides.py:101]: use tempfile here now.

File: src/lamp_py/ingestion/glides.py [hunk 3]
```diff
@@ -217,7 +220,7 @@ def event_schema(self) -> pyarrow.schema:
                                     [("badgeNumber", pyarrow.string())]
                                 ),
                             ),
-                            ("signed_in_at", pyarrow.timestamp("ms")),
+                            ("signedInAt", pyarrow.timestamp("ms")),
```
Comment (mzappitello) [inline src/lamp_py/ingestion/glides.py:223]: this field was incorrectly labeled, so i wasn't picking anything up. :/

File: src/lamp_py/ingestion/glides.py [hunk 4]
```diff
@@ -244,6 +247,11 @@ def unique_key(self) -> str:
         return "operator"
 
     def convert_records(self) -> pd.Dataset:
+        for record in self.records:
+            record["data"]["signedInAt"] = datetime.fromisoformat(
+                record["data"]["signedInAt"].replace("Z", "+00:00")
+            )
```
Comment (mzappitello) [inline src/lamp_py/ingestion/glides.py:253]: i have to do this if we want the record to be stored as a pyarrow.timestamp.

would also do a string if you prefer.

Comment (rymarczy) [reply src/lamp_py/ingestion/glides.py:253]: String probably makes more sense for these, doing the conversion later is pretty inconsequential for such a small dataset

**Review by rymarczy — APPROVED** [review]
LGTM :cake:

**Comment by mzappitello** [comment]
i'll flip that to a string real quick.

**Review by rymarczy — APPROVED** [review]
:cake:

**Pull request merged**